cv_next/lib/utils.ts

```diff
@@ -44,5 +45,61 @@ export function encodeValue(value: string | undefined) {
   return buff.toString("base64");
 }
 
+/**
+ * Creates an Ok Result object with the specified value.
+ *
+ * @param {T} val - the value to be wrapped in the Result object
+ * @return {Result<T, never>} the Result object containing the specified value
+ */
+export function Ok<T>(val: T): Result<T, never> {
+  return { ok: true, val };
+}
+
+export namespace Ok {
+  export const EMPTY = Ok<undefined>(undefined);
+}
+
+/**
+ * Creates an Error Result object with the specified error value.
+ *
+ * @param {E} val - the error value
+ * @return {Result<never, E>} the Result object with ok set to false and containing the specified error value
+ */
+export function Err<E>(
+  where: E,
+  postgrestError?: PostgrestError,
+  err?: Error
+): Result<never, E> {
+  return { ok: false, where, postgrestError, err };
+}
+
+/**
+ * Checks if 2 arrays have the same values in them
+ *
+ * @param {number[]} arr1 - the first array
+ * @param {number[]} arr2 - the second array
+ * @return {boolean} true if arrays have the same values, false otherwise
+ */
+export function arraysHaveSameContent(arr1: number[], arr2: number[]) {
```

Comment:
It's called deep-comparison, and it'd be better if you'd take a pre-made function (e.g. from lodash), as they handle more use-cases :)